app/src/main/java/com/example/happymeals/userlogin/LoginActivity.java

```diff
@@ -100,7 +98,9 @@ public void onFailure(Exception e) {
         register.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                startActivity(new Intent(Login.this, Register.class));
+                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
+                // close this activity
+                finish();
```

Comment:
Have you debugged this line of code? Does this get called when you expect it to?